### PR TITLE
Add correct response for OCS*Exceptions

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Based on #24

### CODE REVIEW

File: src/ControllerMethod.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public static function parse(string $context, array $definitions, ClassMethod $m
 							} else {
 								$responseDescriptions[$statusCode] = $docNode->value->description;
 							}
-							$responses[] = new ControllerMethodResponse($docNode->value->type, $statusCode, "text/plain", new OpenApiType(type: "string"), null);
+							$responses[] = new ControllerMethodResponse($docNode->value->type, $statusCode, "application/json", new OpenApiType(type: "array", maxLength: 0), null);
```
Comment (provokateurin) [inline src/ControllerMethod.php:85]: This change should only be applied when a OCSException was thrown: https://github.com/nextcloud/server/blob/ff2100a4e515988d3ba44c7cfe734d19a6b4b377/lib/private/AppFramework/Middleware/OCSMiddleware.php#L78
In all other cases the error still remains a text/plain string.

File: tests/lib/Controller/SettingsController.php
Comment (provokateurin) [inline tests/lib/Controller/SettingsController.php:1]: Add a test case for non OCSExceptions.

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: src/ControllerMethod.php (hunk 1 above)
Comment (nickvergessen) [reply src/ControllerMethod.php:85]: Well the problem is it's `instanceof` in PHP, that means inheritance works, but it does not work here as we don't load any code...

Comment (nickvergessen) [reply src/ControllerMethod.php:85]: right, but should document this somewhere at some point...

Comment (provokateurin) [reply src/ControllerMethod.php:85]: Yeah, but that is general OCS documentation and not specific to OpenAPI

**Review by provokateurin — APPROVED** [review]

**Pull request merged**